src/settings/MappingProfiles/detailsSections/edit/utils.js

```diff
@@ -0,0 +1,32 @@
+import { get } from 'lodash';
+
+export const onAdd = (refTable, fieldName, incrementalField, fieldIndex, initialFields, callback) => {
+  const fieldsPath = `profile.mappingDetails.mappingFields[${fieldIndex}].subfields`;
+  let newInitRow = { ...get(initialFields, [fieldName], {}) };
+
+  if (incrementalField) {
+    newInitRow = {
+      ...newInitRow,
+      [incrementalField]: refTable.length,
+    };
+  }
+
+  refTable.push(newInitRow);
+  callback(fieldsPath, refTable);
+};
+
+export const onRemove = (index, refTable, incrementalField, fieldIndex, callback) => {
```

Comment:
If `incrementalField` coudn't be set into the function may it's better to make it the last argument of this function?